src/shared_utilities/FileReader.jl

```diff
@@ -19,34 +19,82 @@ using CFTime
 
 using ClimaLSM.Regridder
 
-export to_datetime,
+export AbstractPrescribedData,
+    PrescribedDataTemporal,
+    PrescribedDataSpatial,
     FileInfo,
     FileState,
     SimInfo,
-    PrescribedData,
-    prescribed_data_init,
     read_data_fields!,
     next_date_in_file,
-    interpolate_data
+    interpolate_data,
+    to_datetime
 
 
+"""
+    abstract type AbstractPrescribedData
+
+An abstract type for storing prescribed data info. Subtypes
+include temporally-varying prescribed data and static prescribed data.
+"""
+abstract type AbstractPrescribedData end
```

Comment:
looks good! I like the idea of eventually having methods for each concrete type of this for `next_field!` and `set_initial_field!` (similar to what we do with the albedo now, where `next_field!` doenst update anything in one case, but does in another. not sure how that will play out, just an idea for now!